bundles/org.eclipse.equinox.http.jetty/src/org/eclipse/equinox/http/jetty/internal/HttpServerManager.java

```diff
@@ -285,22 +285,13 @@ public static class InternalHttpServiceServlet implements HttpSessionListener, H
 		// private static final long serialVersionUID = 7477982882399972088L;
 		private final Servlet httpServiceServlet = new HttpServiceServlet();
 		private ClassLoader contextLoader;
-		private final Method sessionDestroyed;
-		private final Method sessionIdChanged;
+		
+		//removing the final keyword
+		private Method sessionDestroyed;
+		private Method sessionIdChanged;
 
 		public InternalHttpServiceServlet() {
-			Class<?> clazz = httpServiceServlet.getClass();
-
-			try {
-				sessionDestroyed = clazz.getMethod("sessionDestroyed", new Class<?>[] { String.class }); //$NON-NLS-1$
-			} catch (Exception e) {
-				throw new IllegalStateException(e);
-			}
-			try {
-				sessionIdChanged = clazz.getMethod("sessionIdChanged", new Class<?>[] { String.class }); //$NON-NLS-1$
-			} catch (Exception e) {
-				throw new IllegalStateException(e);
-			}
+			//removed the constructor code because it wasn't correct , method object was created using reflaction but wrong parameters were getting passed to it.
```

Comment:
Delaying the problem won't help.  These methods must exist or there must be an alternatives that exist in their place.   Moreover, the sample code is using Jetty 9.4.x while Eclipse is using 12.x with ee8, so that seems completely unworkable to me as well.  I tried to get the thing to even compile without errors with no luck.  Before we can informatively discuss what can be changed, there needs to exist something that actually runs and demonstrates the problem are trying to fix.